src/Microsoft.IdentityModel.Tokens/TokenUtilities.cs

```diff
@@ -150,10 +150,25 @@ internal static IEnumerable<SecurityKey> GetAllSigningKeys(TokenValidationParame
             if (validationParameters.IssuerSigningKeys != null)
                 foreach (SecurityKey key in validationParameters.IssuerSigningKeys)
                     yield return key;
+        }
 
-            if (validationParameters.Configuration?.SigningKeys != null)
-                foreach (SecurityKey key in validationParameters.Configuration.SigningKeys)
-                    yield return key;
+
+        /// <summary>
+        /// Returns all <see cref="SecurityKey"/> provided in <paramref name="configuration"/> and <paramref name="configuration"/>.
+        /// </summary>
+        /// <param name="validationParameters">A <see cref="TokenValidationParameters"/> required for validation.</param>
+        /// <param name="configuration">The <see cref="BaseConfiguration"/> that contains signing keys used for validation.</param>
+        /// <returns>Returns all <see cref="SecurityKey"/> provided in provided in <paramref name="configuration"/> and <paramref name="configuration"/></returns>
+        internal static IEnumerable<SecurityKey> GetAllSigningKeys(TokenValidationParameters validationParameters, BaseConfiguration configuration)
```

Comment:
Similar to ResolveTokenSigningKey(), I'm not sure if this method really needs an overload.

Reply:
Modified again in this PR: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/1786/files